src/test/bridges/yearn/YearnUnit.t.sol

```diff
@@ -0,0 +1,210 @@
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 1, 0, 2, address(0));
+    }
+
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testYearnBridgeUnitTestERC20(uint96 _depositAmount) public {
+        IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+        address vault = _registry.latestVault(address(DAI));
+        uint256 availableDepositLimit = IYearnVault(vault).availableDepositLimit();
+        vm.assume(_depositAmount > 1 && _depositAmount < availableDepositLimit);
+
+        deal(address(DAI), address(bridge), _depositAmount);
+        uint256 daiBalanceBefore = DAI.balanceOf(address(bridge));
+
+        AztecTypes.AztecAsset memory depositInputAssetA = AztecTypes.AztecAsset({
+            id: 1,
+            erc20Address: address(DAI),
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+        AztecTypes.AztecAsset memory depositOutputAssetA = AztecTypes.AztecAsset({
+            id: 1,
+            erc20Address: address(vault),
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        // Deposit
+        (uint256 outputValueA, , ) = bridge.convert(
+            depositInputAssetA,
+            emptyAsset,
+            depositOutputAssetA,
+            emptyAsset,
+            _depositAmount,
+            0,
+            0,
+            address(0)
+        );
+        uint256 daiBalanceMid = DAI.balanceOf(address(bridge));
+        uint256 shareAmountMid = IERC20(address(vault)).balanceOf(address(bridge));
+        assertGt(outputValueA, 0, "Share receive should not be 0");
+        assertEq(shareAmountMid, outputValueA, "Invalid share amount");
+        assertLt(daiBalanceMid, daiBalanceBefore, "DAI balance after is not less than before");
+
+        // Withdraw
+        (uint256 outputValueA2, , ) = bridge.convert(
+            depositOutputAssetA,
+            emptyAsset,
+            depositInputAssetA,
+            emptyAsset,
+            outputValueA,
+            0,
+            1,
+            address(0)
+        );
+        uint256 daiBalanceAfter = DAI.balanceOf(address(bridge));
+        uint256 shareAmountAfter = IERC20(address(vault)).balanceOf(address(bridge));
+        assertGt(outputValueA2, 0, "DAI receive should not be 0");
+        assertLt(shareAmountAfter, shareAmountMid, "Invalid share amount");
+        assertGt(daiBalanceAfter, daiBalanceMid, "DAI balance after is not more than before");
+    }
+
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testYearnBridgeUnitTestETH(uint96 _depositAmount) public {
+        IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+        address vault = _registry.latestVault(address(WETH));
+        uint256 availableDepositLimit = IYearnVault(vault).availableDepositLimit();
+        vm.assume(_depositAmount > 1 && _depositAmount < availableDepositLimit);
```

Comment:
You can use `bound` from the test contract instead. This might run into issues if the depositLimit is small as assume will fail a lot.